app/controller/app.js

```diff
@@ -107,6 +109,50 @@ export const search = ({
     return next();
 });
 
+export const markFavorites = ({
+    favoriteProgramsCollection = required('favoriteProgramsCollection'),
+    logger = required('logger', 'You must pass a logging instance for this function to use')
+}) => coroutine(function* (req, res, next){
+    const {
+        user
+    } = req;
+
+    if (!user) {
+        // No need to continue
+        return next();
+    }
+
+    let favorites = null;
+
+    try {
+        favorites = yield getFavoriteProgramsForUser({
+            favoriteProgramsCollection,
+            userId: user._id
+        });
+    } catch (e) {
+        logger.error(e, 'Error finding favorite programs for user');
```

Comment:
Should return next here since we cannot recover and is not critical to showing results